### PR TITLE
Flatten ResponseInfo

### PR DESCRIPTION
**Pull request by jaekwon**
* Flattens ResponseInfo fields
* Changes the return type in types.Application/EndBlock() to ResponseEndBlock
* Some fixes to clean up after tests

### CODE REVIEW

File: types/messages.go [hunk 1]
```diff
@@ -93,9 +93,10 @@ func ToResponseFlush() *Response {
 	}
 }
 
-func ToResponseInfo(info string, tmspInfo *TMSPInfo, blockInfo *LastBlockInfo, configInfo *ConfigInfo) *Response {
+func ToResponseInfo(resInfo ResponseInfo) *Response {
+	resInfoCopy := resInfo
```
Comment (ebuchman) [inline types/messages.go]: are these copies really necessary?

**Comment by ebuchman** [comment]
LGTM otherwise but not tested. maybe let's get the corresponding changes on `tendermint` first before merging?

File: types/messages.go (hunk 1 above)
Comment (jaekwon) [inline types/messages.go]: No, for some reason I thought it was.  I'll remove.

File: types/types.proto [hunk 2]
```diff
@@ -165,11 +165,10 @@ message ResponseFlush{
 }
 
 message ResponseInfo {
-	string info = 1; // backwards compatible
-
-	TMSPInfo tmsp_info = 2;
-	LastBlockInfo last_block = 3;
-	ConfigInfo config = 4;
+	string data = 1;
+	string version = 2;
```
Comment (jaekwon) [inline types/types.proto]: How does it get lower?  Tendermint should try to get some info from the app, and InfoSync() is how Tendermint gets that info during handshake.

Comment (ebuchman) [inline types/types.proto]: It could be handled tmsp/client and tmsp/server instead of the app/tendermint

**Pull request merged**